src/commands/friends/FriendCommand.ts

```diff
@@ -13,3 +11,159 @@ friend list [@Joker#3650]
+          prompt: 'Please provide a valid argument for the used subcommand.',
+          default: '',
+          type: 'user|string'
+        }
+      ]
+    })
+  }
+
+  async run(
+    msg: CommandMessage,
+    {
+      subcommand,
+      argument
+    }: { subcommand: string; argument: User | 'incoming' | 'outgoing' }
+  ): Promise<Message | Message[]> {
+    switch (subcommand.toLowerCase()) {
+      case 'add':
+        return this.sendFriendRequest(msg, argument)
+
+      case 'deny' || 'decline':
+        return this.denyFriendRequest(msg, argument)
+
+      case 'accept':
+        return this.acceptFriendRequest(msg, argument)
+
+      case 'remove' || 'delete':
+        return this.deleteFriend(msg, argument)
+
+      case 'list':
+        return this.listFriends(msg, argument)
+
+      case 'requests':
+        return this.listFriendRequests(msg, argument)
+
+      default:
+        return msg.reply(`\`${subcommand}\` is not a valid subcommand.`)
+    }
+  }
+
+  async sendFriendRequest(
+    msg: CommandMessage,
+    argument: User | 'incoming' | 'outgoing'
+  ): Promise<Message | Message[]> {
+    const receiver = await this.getApiUser(argument)
+    const sender = await this.getApiUser(msg.author)
+
+    if (receiver === null) {
+      return msg.reply(
+        'This command requires you to specify a user. Please try again.'
+      )
+    }
+
+    await axios
+      .post(`http://localhost:5000/api/users/${receiver.id}/friends/requests`, {
+        user: sender,
+        receiver: receiver
+      })
+      .catch(Logger.error)
+
+    return msg.reply(`Sent a friend request to **${receiver.name}**.`)
+  }
+
+  async denyFriendRequest(
+    msg: CommandMessage,
+    argument: User | 'incoming' | 'outgoing'
+  ): Promise<Message | Message[]> {
+    // TODO: Delete friend request using API.
+    return msg.reply('This command is not ready yet.')
+  }
+
+  async acceptFriendRequest(
+    msg: CommandMessage,
+    argument: User | 'incoming' | 'outgoing'
+  ): Promise<Message | Message[]> {
+    // TODO: Create friend using API (no need to delete the request).
+    return msg.reply('This command is not ready yet.')
+  }
+
+  async deleteFriend(
+    msg: CommandMessage,
+    argument: User | 'incoming' | 'outgoing'
+  ): Promise<Message | Message[]> {
+    // TODO: Delete friend using API.
+    return msg.reply('This command is not ready yet.')
+  }
+
+  async listFriends(
+    msg: CommandMessage,
+    argument: User | 'incoming' | 'outgoing'
+  ): Promise<Message | Message[]> {
+    // TODO: List friends using API.
+    return msg.reply('This command is not ready yet.')
+  }
+
+  async listFriendRequests(
+    msg: CommandMessage,
+    argument: User | 'incoming' | 'outgoing'
+  ): Promise<Message | Message[]> {
+    // TODO: List friend requests using API.
+    return msg.reply('This command is not ready yet.')
+  }
+
+  async getApiUser(user: any): Promise<NightwatchUser | null> {
+    if (user instanceof User) {
```

Comment:
Replace the function body with
```ts
const { data } = await axios.get(`{config.api.address}/users/${id}?token=${config.api.token}`)
return data
```

You'll need to get the config from the Plugin class.